src/llmcompressor/sentinel.py

```diff
@@ -0,0 +1,53 @@
+import inspect
+
+from pydantic_core import core_schema
+
+_registry = {}
+
+
+class Sentinel:
+    """
+    Unique sentinel values. Implements https://peps.python.org/pep-0661/
```

Comment:
probably worth calling out the pydantic suggestion since this implementation does not use the `sentinel` package referenced in the PEP

https://github.com/pydantic/pydantic/discussions/9943#discussioncomment-10131374